### PR TITLE
[server] Issue JWT cookie in IAM Sessions for OIDC sign-in WEB-416

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

This change ensures we forward all cookies from IAM sessions to the client

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-server-460cfce934</li>
	<li><b>🔗 URL</b> - <a href="https://mp-server-460cfce934.preview.gitpod-dev.com/workspaces" target="_blank">mp-server-460cfce934.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [x] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>
 
/hold

### CODE REVIEW

File: components/server/src/session-handler.ts [hunk 1]
```diff
@@ -124,6 +124,23 @@ export class SessionHandlerProvider {
...
+    ): Promise<{ name: string; value: string; opts: express.CookieOptions }> {
+        const token = await this.authJWT.sign(userID, {});
+
+        return {
+            name: SessionHandler.getJWTCookieName(this.config),
+            value: token,
+            opts: {
+                maxAge: this.config.auth.session.cookie.maxAge * 1000, // express does not match the HTTP spec and uses milliseconds
+                httpOnly: this.config.auth.session.cookie.httpOnly,
+                sameSite: this.config.auth.session.cookie.sameSite,
```
Comment (geropl) [inline components/server/src/session-handler.ts:138]: Somehow seeing this being configurable makes me uneasy... :worried: But I recall we discussed before, and I assume we had good reasons. :ok_hand:

Comment (easyCZ) [reply components/server/src/session-handler.ts:138]: The cookie config has to be consistent between public-api and server. That means the only way to ensure that is that we share the same config. It's not configurable from the `installer` side, but both public-api and server components use the same underlying config

**Comment by geropl** [comment]
I tested this, but somehow ended up in a super weird state. We're now trying to re-produce with a clean slate.

**Comment by easyCZ** [comment]
Screenshot 2023-05-30 at 16 24 30

Can confirm I got the JWT cookie after performing the SSO sign-in

**Comment by easyCZ** [comment]
Used preview from this branch: https://github.com/gitpod-io/gitpod/pull/17784 which is the same branch, except has a working preview env unlike this PR (which got wrecked somehow)

**Pull request merged**